### PR TITLE
Add BrowserFetcher bindings

### LINKED ISSUE #40: Bindings for BrowserFetcher
https://github.com/GoogleChrome/puppeteer/blob/v1.2.0/docs/api.md#class-browserfetcher

### PR DESCRIPTION
**Pull request by jihchi**
Fix #40 

--

These bindings increase our unit test duration.
screenshot_2018-03-26_22-41-41_ps9gx

### CODE REVIEW

**Comment by zploskey** [comment]
These look good to me. It's debatable whether we should bother typing the `platform` return value beyond a string. I doubt this will be used often. Would be fine with it as is.

File: src/BrowserFetcher.re [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  download(t, revision, progressCallback |> Js.Undefined.fromOption);
+
+[@bs.send] external localRevisions : t => Js.Promise.t(array(string)) = "";
+
+[@bs.deriving jsConverter]
+type platform = [
+  | [@bs.as "mac"] `Mac
+  | [@bs.as "linux"] `Linux
+  | [@bs.as "win32"] `Win32
+  | [@bs.as "Win64"] `Win64
```
Comment (zploskey) [inline src/BrowserFetcher.re]: win64 should be all lower case. If you just make the polymorphic variants all lower case and the same as the string doesn't it convert them to the right string? No need to use bs.as.

Comment (jihchi) [reply src/BrowserFetcher.re]: I thought variant constructor need to be capitalized. tried lower case but without success.

Comment (zploskey) [reply src/BrowserFetcher.re]: Polymorphic variants don't need to be capitalized. For example: https://github.com/bs-puppeteer/bs-puppeteer/blob/master/src/ConsoleMessage.re

Comment (jihchi) [reply src/BrowserFetcher.re]: Oh! That works, Thanks!

**Comment by jihchi** [comment]
Hi @zploskey ,

PR is done.

File: src/Puppeteer.re [hunk 2]
```diff
@@ -116,3 +116,32 @@ external launch : (~options: launchOptions=?, unit) => Js.Promise.t(Browser.t) =
...
+
+type browserFetcherOptions = {
+  .
+  "host": Js.undefined(string),
+  "path": Js.undefined(string),
+  "platform": Js.undefined(string),
+};
+
+[@bs.obj]
+external makeBrowserFetcherOption :
```
Comment (zploskey) [inline src/Puppeteer.re]: To be consistent this should be `makeBrowserFetcherOptions`, plural.

Comment (jihchi) [reply src/Puppeteer.re]: Fixed in commit 784ba5c

**Pull request merged**